src/libcore/fmt/mod.rs

```diff
@@ -1383,7 +1383,7 @@ impl Debug for str {
         f.write_char('"')?;
         let mut from = 0;
         for (i, c) in self.char_indices() {
-            let esc = c.escape_default();
+            let esc = c.escape();
```

Comment:
These should be updated as well I think?

(may be good to do a smoke test with `make check`